internal/fabric/client/eventsubscriber.go

```diff
@@ -89,25 +89,29 @@ func (e *eventClientWrapper) subscribeEvent(subInfo *eventsapi.SubscriptionInfo,
 	}
 }
 
-func (e *eventClientWrapper) getEventClient(channelId, signer string, since uint64) (eventClient *event.Client, err error) {
+func (e *eventClientWrapper) getEventClient(channelId, signer string, since uint64, chaincodeId string) (eventClient *event.Client, err error) {
 	eventClientsForSigner := e.eventClients[signer]
 	if eventClientsForSigner == nil {
 		eventClientsForSigner = make(map[string]*event.Client)
 		e.eventClients[signer] = eventClientsForSigner
 	}
-	eventClient = eventClientsForSigner[channelId]
+	key := eventsapi.GetKeyForEventClient(channelId, chaincodeId)
```

Comment:
since this is an in-memory array, it'll be re-built with the new key scheme. But you question does apply to the `submanager.go` that handles the lookup entry when adding or deleting subscriptions. A possible glitch is when a subscription is created that conflicts with an existing subscription created in previous versions, it will be allowed to be created (and it won't work).

This is already the case with the code right now. The user can observe the logs (about failing to register the new listener) and delete the subscription.